### PR TITLE
Add recipe for eightpoints/guzzle-bundle

### PR DESCRIPTION
**Pull request by gregurco**
| Q             | A
| ------------- | ---
| License       | MIT

This bundle integrates Guzzle 6.x into Symfony. Guzzle is a PHP framework for building RESTful web service clients.

### CODE REVIEW

File: eightpoints/guzzle-bundle/7.0/config/packages/dev/eight_points_guzzle.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+eight_points_guzzle:
+    logging: true
```
Comment (Nyholm) [inline eightpoints/guzzle-bundle/7.0/config/packages/dev/eight_points_guzzle.yaml]: The default value for this parameter is KERNEL_DEBUG, right?

File: eightpoints/guzzle-bundle/7.0/config/packages/eight_points_guzzle.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+eight_points_guzzle:
+    clients: ~
```
Comment (Nyholm) [inline eightpoints/guzzle-bundle/7.0/config/packages/eight_points_guzzle.yaml]: Is this really needed? I just looked at the code quickly, it seams like the bundle will have the exact same behaviour without these two lines, is that correct?

File: eightpoints/guzzle-bundle/7.0/config/packages/dev/eight_points_guzzle.yaml (hunk 1 above)
Comment (gregurco) [reply eightpoints/guzzle-bundle/7.0/config/packages/dev/eight_points_guzzle.yaml]: yep, `%kernel.debug%`

File: eightpoints/guzzle-bundle/7.0/config/packages/eight_points_guzzle.yaml (hunk 2 above)
Comment (gregurco) [reply eightpoints/guzzle-bundle/7.0/config/packages/eight_points_guzzle.yaml]: yep, behaviour the same, but user will know where to add clients...

**Comment by gregurco** [comment]
@Nyholm I added client in config file, good idea :+1:

File: eightpoints/guzzle-bundle/7.0/config/packages/eight_points_guzzle.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+eight_points_guzzle:
+    clients:
+        api_payment:
```
Comment (Nyholm) [inline eightpoints/guzzle-bundle/7.0/config/packages/eight_points_guzzle.yaml]: Rename to ‘my_client’

File: eightpoints/guzzle-bundle/7.0/config/packages/eight_points_guzzle.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+            base_url: "http://api.domain.tld"
```
Comment (Nyholm) [inline eightpoints/guzzle-bundle/7.0/config/packages/eight_points_guzzle.yaml]: Remove

File: eightpoints/guzzle-bundle/7.0/config/packages/eight_points_guzzle.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+            options:
+                timeout: 30
+
+                headers:
+                    Accept: "application/json"
```
Comment (Nyholm) [inline eightpoints/guzzle-bundle/7.0/config/packages/eight_points_guzzle.yaml]: Add a comment before this line to explain what we are doing. 
Maybe aleo change this to an less important header. Like the User-Agent header.

File: eightpoints/guzzle-bundle/7.0/config/packages/eight_points_guzzle.yaml (hunk 4 above)
Comment (gregurco) [reply eightpoints/guzzle-bundle/7.0/config/packages/eight_points_guzzle.yaml]: Better to keep here some URL. Probably one 100% not active or to write `http://target.url`. It's better to enforce users to write host here, not in controllers and it's better to find proper way to do that

File: eightpoints/guzzle-bundle/7.0/config/packages/eight_points_guzzle.yaml (hunk 5 above)
Comment (gregurco) [reply eightpoints/guzzle-bundle/7.0/config/packages/eight_points_guzzle.yaml]: wrote comment

**Comment by Nyholm** [comment]
The point of the recipes is to bring the users good default configuration that will work for 90% of the users. When we have a base_url, the user cannot start using the new services that the bundle provides without editing the configuration. 
Same with the Accept header. I think it is important to show how to configure headers by example like you are doing. But specifying "Accept: application/json" it will not be working for 90% of the cases. I still suggest to use the "User-Agent" header.

That is just my 50 cent.

**Comment by gregurco** [comment]
@Nyholm ok, but which one to use? `test/1.0`, `EightpointsGuzzleBundle/1.0` or another one?

**Comment by Nyholm** [comment]
I think `EightpointsGuzzleBundle/7.0` would be a good User-Agent.

File: eightpoints/guzzle-bundle/7.0/config/packages/eight_points_guzzle.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+eight_points_guzzle:
+    clients:
+        my_client:
+            # Write here the host where to do requests
+            base_url: "http://target.url"
```
Comment (Nyholm) [inline eightpoints/guzzle-bundle/7.0/config/packages/eight_points_guzzle.yaml]: You like this line and I agree it is important. But lets make it a comment and this service will be useable without the user has to change the config.

Comment (gregurco) [reply eightpoints/guzzle-bundle/7.0/config/packages/eight_points_guzzle.yaml]: A leave `base_url` here but commented it. Is it ok?

**Pull request merged**